addons/mrp_subcontracting/models/stock_warehouse.py

```diff
@@ -41,6 +53,14 @@ def get_rules_dict(self):
             })
         return result
 
+    def _update_global_route_resupply_subcontractor(self):
+        route_id = self._find_global_route('mrp_subcontracting.route_resupply_subcontractor_mto',
+                                           _('Resupply Subcontractor on Order'))
+        if not route_id.rule_ids:
+            route_id.active = False
+        else:
+            route_id.active = True
```

Comment:
You need to search for all company for rules that are assigned to this route. Right now if you are in a multi-company situation and only have 1 company selected then if you deactivate all of its warehouses' "Resupply Subcontractors" then it will archive the route even if the other companies still have warehouses with "Resupply Subcontractors" = True